### DISCUSSION TITLE
Any async commands will not work after a while

### DISCUSSION THREAD

**Discussion by ir210** [minad/consult#937] [Q&A] [answered]
Hello,

I have been facing an issue for a few months now where async commands work initially but stop showing any results interactively after a while. By async commands, I mean those using `(consult--with-async)`. If I make changes to the prompt, the result list doesn't update. I am not sure what causes this issue. However, restarting Emacs temporarily fixes the issue.

I have narrowed down the issue to the `after-change-functions` hook not being called at all when the behavior is observed. However, when I run the hook manually during a `consult--read` session (by adding a new custom command), it works. It retriggers the async chain and updates the list. I have also checked that the hook is not removed from the hook list.

Since nobody else has reported this error, I think it's only happening on my local machine. I am sharing my configuration details here, hoping it might be useful:

- Emacs 29.2
- Spacemacs (always kept up-to-date)
- Vertico
- ripgrep 14.1.0 (I mostly use `consult-ripgrep`, but any async command behaves the same)

I would really appreciate any help or pointers on how to fix this issue. I heavily rely on `(consult-ripgrep)` to perform my daily job.

**Comment by minad** [comment]
Hi! Personally I've not observed such issues. In order to debug this problem I need a minimal recipe starting from emacs -Q (not spacemacs), see https://github.com/minad/consult?tab=readme-ov-file#bug-reports.

**Reply by ir210** [reply]
I'll try doing that, but it might take a while before I can come up with a minimal setup that reproduces it. Even in my current setup, it still works after a restart. Only when I did something at some point, the behavior started to show up. The thing is, I am unable to identify the specific action that triggers it.

I am wondering why Emacs doesn't trigger the `after-change-functions` hook when the prompt is changed. Initially, I thought it might be a bug in Emacs, but I noticed that it only happens when running the async commands. The non-async consult commands work fine. Do you know if there is any case that prevents the hook from being triggered even when there is a change in the minibuffer? I can't figure it out from the documentation.

**Reply by ir210** [reply]
Turned out `inhibit-modification-hooks` is `t`. That's why the hooks are not run. Now, I need to find out what sets it to `t`.

**Reply by ir210** [reply, marked as answer]
As I suspected, my own package mistakenly sets `inhibit-modification-hooks` to `t`. Maybe I wanted to experiment with something but forgot to remove it 😅. Anyway, thank you for responding. As usual, talking to other people always helps.
